### PR TITLE
Added new error to raw sending API - MissingFromAddress as distinct from UnauthenticatedFromAddress

### PR DESCRIPTION
**Pull request by matthewbloch**
This new error would have saved me a couple of hours understanding what I was doing wrong in this issue: https://github.com/atech/postal/issues/530

### CODE REVIEW

**Comment by willpower232** [comment]
Please could you change it to "FromAddressMissing" for parity with the exception defined here: https://github.com/atech/postal/blob/8b7bcd9a6a3f68051f76e81385f39ede1850e4f6/api/controllers/send_api_controller.rb#L30

**Comment by willpower232** [comment]
Thanks, hopefully someone from atech can get it merged soon along with the others

File: api/controllers/send_api_controller.rb [hunk 1]
```diff
@@ -84,7 +85,11 @@
 
       # If we're not authenticated, don't continue
       if authenticated_domain.nil?
-        error 'UnauthenticatedFromAddress'
+        if mail.from.nil? || mail.from.empty? || !/@/.match(mail.from.first)
```
Comment (catphish) [inline api/controllers/send_api_controller.rb:88]: I would suggest removing the regex check. The from address is not "missing" if it's merely invalid. Also, I assume we have Rails extensions and could therefore just use `mail.from.blank?`

**Review by catphish — CHANGES_REQUESTED** [review]

**Comment by adamcooke** [comment]
I'm planning an API rewrite and am now reluctant to change the behaviour of the old one. The new one will include something like this.

**Pull request closed without merging**